cinema/views.py

```diff
@@ -56,3 +93,66 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            date = parse_date(date)
+            if date:
+                queryset = queryset.filter(show_time__date=date)
+
+        if movie:
+            queryset = queryset.filter(movie_id=movie)
+
+        if self.action in "list":
+            return (
+                queryset
+                .select_related("movie", "cinema_hall")
```

Comment:
In the `MovieSessionViewSet`, using `select_related()` without specifying fields can lead to unnecessary database joins. Consider specifying the fields to optimize the query.